src/pages/[userId]/overlays/prediction.tsx

```diff
@@ -221,67 +217,64 @@ export default function Page() {
   const subscribeToPredictions = api.subscriptions.predictions.useMutation();
   const { lastMessage } = useWebSocket(socketUrl);
 
-  const handleSubscribingToPredictions = async (
-    userId: string,
-    sessionId: string,
-  ) => {
-    await subscribeToPredictions.mutateAsync({
-      userId,
-      sessionId,
-    });
-  };
-
-  const handleReconnectUrl = async (reconnect_url: string | null) => {
-    setSocketUrl(reconnect_url ?? socketUrl);
-  };
+  useEffect(() => {
+    const handleSubscribingToPredictions = async (
+      userId: string,
+      sessionId: string,
+    ) => {
+      await subscribeToPredictions.mutateAsync({
+        userId,
+        sessionId,
+      });
+    };
+    
+    const handleReconnectUrl = async (reconnect_url: string | null) => {
+      setSocketUrl(reconnect_url ?? socketUrl);
+    };
```

Comment:
I'm not sure I understand why these function definitions should all be moved inside the `useEffect` hook. Could you explain?

Reply:
After re-reading useEffect's docs I think I understand actually, since they are only used within the hook this prevents the functions from getting redefined every re-render and instead only when the dependencies in the array are changed.